### DISCUSSION TITLE
Odd post-assembly problem

### DISCUSSION THREAD

**Discussion by MaggieLeber** [simpleavr/s14clock#2] [Q&A] [answered]
OK, I completed the kit version (and realized, after putting the S2 sockets on the wrong side of the board, how my building skills have deteriorated, so have a completed version on its way too). Meanwhile I have that problem corrected, and solved my missing-digit problems by re-flowing the 74HC154D. 

I have one remaining oddly bad character, and am head-scratching how this could happen. I'm pretty sure that I've got too much solder on several of the 3692AS pins, it having run down the pin to the other side of the board (between the board and the display) without me noticing.  

So what's puzzling me is...I think I have the operation of the multiplexed display figured out; the D* lines select the digit position and the common `ABCDEF12HJKLMN dp` lines light up the segments on the selected digit.  

But I have one position where the `hkml` lines don't light up for X, or for / , and yet `hl` DO light up for \\.  

I'd like to minimize the rework I do, because I may be as likely to do more damage as to repair anything. But my brain is broken figuring out where a short/open could be that would cause this specific behavior. 

Video of it happening here: 

https://photos.app.goo.gl/Nu49yjkBYXPFxsUU7

I could understand if `hkml` were ALWAYS dead. But how do `hl` work for reverse slant but not forward?   
I'm better at software than hardware. :-)

**Comment by simpleavr** [comment]
Interesting. Seldom see something like this. Which doesn't seem to make sense.
You would think that if there is internal bridge on the 4th LED module, with the multiplexing scheme, it would impact all other digits at the same segments.
And still, it is more probable that only the 4th LED module is failing with internal solder bridges turning on / off segments.
I would try the following and hope it will show more clues;
1. writeString() to just the 20th (problematic) digit, with all other digits blank, just to see if anything changes.
2. use other combination of affected segments, like '(', ')', '<', '>', and see if there are other on / off combination that can give new clues.

I did encounter something similar that I couldn't figure out, it was a prototype that had a lot of work (i.e. heat to the part). Eventually I decided to replace the LED module. After removing it though, the module was tested and it was good. And it was bent and twisted during removal so I did not re-use it. A new part was put in the fixed the problem.

If I am in your position. I would try to reflow the LED module again, focusing on the problematic segments, and remove excessive solder via solder wick or better yet if you have a pump, if placed properly, they are able to suck solders though the holes.

If this is not working out, I am happy to ship you a replacement LED module. Removing the LED module is quite easy with a pump and some patience, and almost impossible otherwise.

**Comment by MaggieLeber** [comment]
I did do some reflowing work on the module last night. I have a good supply of wick and a decent solder sucker.

I'll get more medieval on it and see what happens. I am encouraged that you are as mystified as I am...makes me feel less dumb.

Thanks for the suggestions...I'll let you know what happens. I plan to do some interesting work on scripting my displays; will share that when I have something.

**Reply by simpleavr** [reply, marked as answer]
I think I figured it out.
I got a unit and turn on the string "XXXXXX...." likewise to test. And I short 'h' and D2 (D20) on the 4th LED module with a piece of wire, and reproduced what you have. Looks like there is a solder bridge between these 2 pins. They are not adjacent to each other (separate by one pin). And if you trace back w/ a multimeter, you should find that the traces are adjacent to each other around the 74HC154D (at pin 17).
Depending on your version of board (likely you got the 1st version), there is a trace between a 0.65mm pins pitch and this is the spot most likely to have a short. I suspect if you rework this area (more flux, wick up addition solders), there is a good chance this can be fixed. On the newer PCB designs, I have re-trace them to avoid running tracks between the 74HC154D pins.

**Reply by MaggieLeber** [reply]
That got it. Didn't even have to rework, just careful cleaning and scrubbing a little between the pins with a cinnamon wood toothpick; what a watchmaker would call a "pegwood". 

Here it is working.... https://photos.app.goo.gl/a8Ka92XbF8Q1HFE5A

**Reply by simpleavr** [reply]
Looks great in front of your keyboard. Hope you enjoy it.

**Comment by MaggieLeber** [comment]
Cool! I'll look into it when I get home. Thanks, and well-spotted..
